### PR TITLE
Add new test based on OpenMEEG failure and existing test in matio

### LINKED ISSUE #38320: shipping  botteled libmatio-1.5.12 compiled against hdf5 1.10.4 not 1.10.5
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [ ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

I am using `matio` client code and I get an error saying that headers and library missmatch. I can reproduce it in several machines (also travis)

### PR DESCRIPTION
**Pull request by massich**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

Fixes #38320

-----

### CODE REVIEW

**Comment by massich** [comment]
@fxcoudert I think this should work as a regression test. Would it show in the CIs? can someone replicate on a mac?

**Comment by massich** [comment]
cc: @agramfort

File: Formula/libmatio.rb [hunk 1]
```diff
@@ -42,6 +42,18 @@ def install
         if (!(matfp = Mat_Open(argv[1], MAT_ACC_RDONLY)))
           abort();
         Mat_Close(matfp);
+
+        mat = Mat_CreateVer("test_writenan.mat",NULL,mat_file_ver);
```
Comment (fxcoudert) [inline Formula/libmatio.rb]: This test cannot pass, whether on macOS or Linux or anywhere else. `mat`, `mat_file_ver`, `matvar` (below), `dims` and `data` are not defined anywhere.

**Comment by massich** [comment]
This uses the same code with the shipped libmatio and fails. 
https://travis-ci.org/massich/ci-sandbox/builds/513103769?utm_source=github_status&utm_medium=notification

Why is this formula passing the CI then?

**Comment by massich** [comment]
@fxcoudert where should we go from here?

**Comment by massich** [comment]
@fxcoudert It still fails, and the test reveals it.

https://travis-ci.org/massich/ci-sandbox/builds/515273516?utm_source=github_status&utm_medium=notification

**Comment by fxcoudert** [comment]
@massich you asked us to amend the test, and the new test is passing on our CI machines. The link you're giving me is outside of Homebrew's control, and I don't actually know what it is.

**Comment by massich** [comment]
The outside link is what I used to do the test that reveals the regression. After you asked for it [here](https://github.com/Homebrew/homebrew-core/issues/38320#issuecomment-477139622) Then I used the code to do this PR as you suggested. What I don't understand is why this did not either fail here or trigger new binaries.

What do you suggest?

**Pull request merged**